esi_leap/objects/lease.py

```diff
@@ -302,8 +302,19 @@ def expire(self, context=None):
                         parent_lease = Lease.get(self.parent_lease_uuid)
                         resource.set_lease(parent_lease)
                 # expire lease
-                self.status = statuses.EXPIRED
-                self.expire_time = datetime.datetime.now()
+                notify.emit_start_notification(context, self,
+                                               'delete', CRUD_NOTIFY_OBJ,
+                                               node=resource)
+                with notify.handle_error_notification(context,
+                                                      self,
+                                                      'delete',
+                                                      CRUD_NOTIFY_OBJ,
+                                                      node=resource):
+                    self.status = statuses.EXPIRED
```

Comment:
Thinking about it, it probably makes the most sense for this `with` block to surround lines 298-303 above, as that's what actually unsets the ironic lessee field. I might consider:

* making this change and also isolating it into its own function, so that here we just call that function and then set the status to EXPIRED and set the expire_time here
* update `cancel` to use this new function
* update `fulfill` to have the `with` block surround the call to `set_lease`

Reply:
Will do!